### PR TITLE
val reducer

### PR DESCRIPTION
**Pull request by Chary-x**
done changes requested

### CODE REVIEW

File: data.py [hunk 1]
```diff
@@ -30,11 +34,20 @@ def load_dataset(partial=True) -> Tuple[DataLoader, DataLoader, DataLoader]:
 
     # Load the full training set & reduce it
     logger.info(f"Loading full training set")
-    full_train_dataset = ImageFolder(os.path.join(DATASET_PATH, 'images/train'), transform=transform)
+    # CHANGE ADDRESS
+    full_train_dataset = ImageFolder(os.path.join('C:\\Users\\Ignatius David\\Desktop\\Datasets\\Plants\\plantnet_300K', 'images/train'), transform=transform)
```
Comment (rsclip) [inline data.py]: Please use `DATASET_PATH` constant here

File: data.py [hunk 2]
```diff
@@ -46,31 +59,56 @@ def load_dataset(partial=True) -> Tuple[DataLoader, DataLoader, DataLoader]:
...
-    val_loader = DataLoader(val_dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=NUM_WORKERS)
+    test_loader = DataLoader(reduced_test_dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=NUM_WORKERS)
+    val_loader = DataLoader(reduced_val_dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=NUM_WORKERS)
 
     return train_loader, test_loader, val_loader
 
 
-# function to reduce the dataset size
-def reduce_dataset(dataset: torch.utils.data.Dataset, samples_per_class: int) -> torch.utils.data.Subset:
+def reduce_dataset(dataset: torch.utils.data.Dataset, samples_per_class: int, random_state=None) -> torch.utils.data.Subset:
```
Comment (rsclip) [inline data.py]: Proper type hints; `random_state: bool = None`

**Review by rsclip — CHANGES_REQUESTED** [review]
The main problem is that the code takes significantly longer. If you're looking for reproducible results, you could attempt that by passing in a seed into `sklearn`'s `train_test_split`. It has a [parameter for random states](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) in the documentation.

In the next commit, complete all changes.

File: data.py (hunk 1 above)
Comment (Chary-x) [reply data.py]: alrr check new commit

File: data.py (hunk 2 above)
Comment (Chary-x) [reply data.py]: done

**Review by rsclip — APPROVED** [review]
`my_app.log` should be removed but I'll take care of that.

Approved 🙂
